### PR TITLE
Feature/dx 246 document paid and failed operations

### PR DESCRIPTION
**Pull request by trieutran91**
Added documentation for `paid` and `failed` operations.

### CODE REVIEW

File: _includes/payment-resource.md [hunk 1]
```diff
@@ -90,6 +90,18 @@ Content-Type: application/json
             "href": "{{ page.apiUrl }}/psp/{{ payment-instrument }}/payments/{{ page.paymentId }}/captures",
             "rel": "create-capture",
             "contentType": "application/json"
+        },
+        {
+            "method": "GET",
+            "href": "{{ page.api.Url }}/psp/{{ payment-instrument }}/{{ page.paymentId }}/paid",
```
Comment (asbjornu) [inline _includes/payment-resource.md]:
```suggestion
            "href": "{{ page.apiUrl }}/psp/{{ payment-instrument }}/{{ page.paymentId }}/paid",
```

File: _includes/payment-resource.md [hunk 2]
```diff
@@ -90,6 +90,18 @@ Content-Type: application/json
...
+            "rel": "paid-payment",
+            "contentType": "application/json"
+        },
+        {
+            "method": "GET",
+            "href": "{{ page.api.Url }}/psp/{{ payment-instrument }}/{{ page.paymentId }}/failed",
```
Comment (asbjornu) [inline _includes/payment-resource.md]:
```suggestion
            "href": "{{ page.apiUrl }}/psp/{{ payment-instrument }}/{{ page.paymentId }}/failed",
```

File: checkout/other-features.md [hunk 3]
```diff
@@ -358,6 +358,18 @@ A list of possible operations and their explanation is given below.
             "href": "{{ page.apiUrl }}/psp/paymentorders/{{ page.paymentOrderId }}/reversals",
             "rel": "create-paymentorder-reversal",
             "contentType": "application/json"
+        },
+        {
+            "method": "GET",
+            "href": "{{ page.api.Url }}/psp/paymentorders/{{ page.paymentOrderId }}/paid",
```
Comment (asbjornu) [inline checkout/other-features.md]:
```suggestion
            "href": "{{ page.apiUrl }}/psp/paymentorders/{{ page.paymentOrderId }}/paid",
```

File: checkout/other-features.md [hunk 4]
```diff
@@ -358,6 +358,18 @@ A list of possible operations and their explanation is given below.
...
+            "rel": "paid-paymentorder",
+            "contentType": "application/json"
+        },
+        {
+            "method": "GET",
+            "href": "{{ page.api.Url }}/psp/paymentorders/{{ page.paymentOrderId }}/failed",
```
Comment (asbjornu) [inline checkout/other-features.md]:
```suggestion
            "href": "{{ page.apiUrl }}/psp/paymentorders/{{ page.paymentOrderId }}/failed",
```

**Comment by trieutran91** [comment]
> The operations are available for absolutely all payment instruments as of now?

It is available for instruments that are used in Checkout. Which leaves out MobilePay, Direct Debit and Credit?

**Comment by asbjornu** [comment]
Right. Then we need some logic in `_includes/payment-resource.md` to differentiate this, no? Perhaps a `showStatusOperations` parameter that is `true` by default?

File: _includes/payment-resource.md [hunk 5]
```diff
@@ -141,6 +156,23 @@ for the given operation.
 
 {% else %}
 
+{% case showStatusOperations %}
+
+{% when true %}
```
Comment (NullableInt) [inline _includes/payment-resource.md]: When true?

File: _includes/payment-resource.md [hunk 6]
```diff
@@ -90,7 +91,21 @@ Content-Type: application/json
             "href": "{{ page.apiUrl }}/psp/{{ payment-instrument }}/payments/{{ page.paymentId }}/captures",
             "rel": "create-capture",
             "contentType": "application/json"
+        },
```
Comment (asbjornu) [inline _includes/payment-resource.md]: This trailing comma should not be here if the `showStatusOperations` is `false`. I think you need to move `
{% if showStatusOperations == true %}` up to line 94 between `}` and `,`.

```suggestion
        }{% if showStatusOperations == true %},
```

File: _includes/payment-resource.md [hunk 7]
```diff
@@ -141,6 +156,23 @@ for the given operation.
 
 {% else %}
 
+{% case showStatusOperations %}
```
Comment (asbjornu) [inline _includes/payment-resource.md]: Are you using `case` to have `else` available? Is `else` not possible to use in conjunction with a regular `if`?

File: payments/mobile-pay/other-features.md [hunk 8]
```diff
@@ -18,6 +18,8 @@ sidebar:
 
 ## Payment Resource
 
+{% include payment-resource.md  payment-instrument="mobilepay" showStatusOperations='false' %}
```
Comment (asbjornu) [inline payments/mobile-pay/other-features.md]: I'm not 100% certain, but I think `false` needs to be passed as a literal and not as a quoted string.

```suggestion
{% include payment-resource.md  payment-instrument="mobilepay" showStatusOperations=false %}
```

**Pull request merged**